### DISCUSSION TITLE
Porting pre-1.4.0 themes to 1.7.0

### DISCUSSION THREAD

**Discussion by DaveDavenport** [davatorium/rofi#1400] [Theming] [answered]
The very old theme format (that was deprecated in 1.4.0 with the introduction of the new theming engine.). There was some 'hackish' conversion code in old rofi versions, that now has been removed as it became to difficult to maintain.

The old format used:  `color-normal`, `color-urgent`, `color-active`, `color-window`. This format was based on i3.

To convert this will take a bit more effort.

**Comment by DaveDavenport** [comment, marked as answer]
If you still use this very old setup and have not upgraded yet, use the following procedure to upgrade.

You need to run 1.6.1 or older to be able to parse the old format.

First copy the old configuration to the new format:
```bash
rofi -dump-config > newconfig.rasi
```
Append the old theme:
```bash
rofi -dump-theme >> newconfig.rasi
```

You can test this using:
```bash
rofi -config newconfig.rasi -show run
```

This might not be a perfect conversion. (see other topics in this discussion form).

Then you can copy the config into `~/.config/rofi/config.rasi` and upgrade rofi.

**Comment by SolsticeSpectrum** [comment]
color-window, color-normal and hide-scrollbar does not exist.

**Reply by DaveDavenport** [reply]
???

**Reply by SolsticeSpectrum** [reply]
It is all different now. Nothing works at all. How do I make rofi fullscreen like it used to be?

**Reply by SolsticeSpectrum** [reply]
Also it has totally different proportions

**Reply by SolsticeSpectrum** [reply]
Not even fullscreen exists

**Reply by DaveDavenport** [reply]
Fullscreen still exists, if you read the rofi-theme manpage you see it is  property on the window.

```css
window {
   fullscreen: true;
}
```

There is a rofi-theme manpage and there have been several topics on this and I suspect the information can be found there.

We might be able to help if you ask a question instead of dropping in random remarks.

**Comment by stevenengler** [comment]
I had luck using an old theme generator at https://comfoxx.github.io/rofi-old-generator/p11-Generator.html, where you can enter the colors from your `color-normal`, `color-urgent`, etc and output a generated rasi config. The only change I also needed was to add the following, for reasons unknown to me.

```
element-text {
    background-color: inherit;
    text-color: inherit;
}
```

**Edit:** Also there's a bug with `alternate-normal-foreground` where it is defined twice, and one of them should have been named `alternate-normal-background` instead. Also there's a `border-color: @foreground` which should have been `border-color: @bordercolor`. You may also want a `#listview { scrollbar: true; }`.
